app/main/views/suppliers.py

```diff
@@ -418,7 +416,13 @@ def edit_supplier_organisation_size():
 @main.route('/trading-status/edit', methods=['GET', 'POST'])
 @login_required
 def edit_supplier_trading_status():
-    form = CompanyTradingStatusForm()
+
+    prefill_data = {}
+    if request.method == "GET":
```

Comment:
Why does this view only conditionally load the existing supplier data and the others load it for both `GET` and `POST`?

Reply:
I was worried about making the GET more expensive; the other views GET was already including the api requests.

I'm happy to take it out though, it is a bit premature optimisation-y

Reply:
I was actually looking at it the other way around - why don't we do it in all the others too? One advantage I can think of is a small amount of added clarity - it's more clear the `prefill_data` can't have any effect in the `POST` case if it never gets filled out in the first place.

Unless, I reasoned, the use of supplier-fetching here is being used as a guard to generate 404s if the supplier doesn't exist - but... that's ... not possible here? Guessing most of this apps use a `login_required` decorator that requires `role == "supplier"`?